compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -532,6 +532,14 @@ impl<'a: 'ast, 'ast> LateResolutionVisitor<'a, '_, 'ast> {
                     enum_candidates.into_iter().map(|(_variant_path, enum_ty_path)| enum_ty_path),
                     Applicability::MachineApplicable,
                 );
+                if self.r.session.features_untracked().associated_const_equality {
+                    err.span_suggestion(
+                        span,
+                        "if intended as const, try wrapping in braces",
+                        format!("{{ {path_str} }}"),
+                        Applicability::MaybeIncorrect,
+                    );
+                }
```

Comment:
Why is this suggestion correct in general?
AFAIK, disambiguating using braces only works for equality associated constraints.
The suggestion needs to be restricted to this case, doesn't it?

Reply:
Do you mean as opposed to other const generics? Disambiguating with braces also works for functions in generic const positions.
By this case, do you mean specifically for when the constraint is a const?

Reply:
I mean as opposed to all the other places where one can put a type.
Does this suggestion fire with this?
```
fn foo() -> Mode::Cool { /* ... */ }
```

Reply:
It does fire, I'll try to see where to change to make it not

Reply:
I've changed it to be only for associated equality, but it's not so specific to enums.